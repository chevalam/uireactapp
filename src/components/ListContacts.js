import React, { Component } from 'react';
import PropTypes from 'prop-types'
import escapeRegExp from 'escape-string-regexp'
import sortBy from 'sort-by'
import MoreVert from 'material-ui-icons/MoreVert';
import IconButton from 'material-ui/IconButton';
import Avatar from 'material-ui/Avatar';
import AccountCircle from 'material-ui-icons/AccountCircle';
import moment from 'moment'
import {grey500} from 'material-ui/styles/colors';

class ListContacts extends Component {
  static propTypes = {
    contacts: PropTypes.array.isRequired
  }

  state = {
    query: ''
  }

  updateQuery = (query) => {
    this.setState({ query: query.trim() })
  }

  clearQuery = () => {
    this.setState({ query: '' })
  }

  render() {
    const { contacts} = this.props
    const { query } = this.state

    let showingContacts
    if (query) {
      const match = new RegExp(escapeRegExp(query), 'i')
      showingContacts = contacts.filter((contact) => match.test(contact.name))
    } else {
      showingContacts = contacts
    }

    showingContacts.sort(sortBy('name'))

    return (
      <div className='list-contacts'>
        <div className='list-contacts-top'>
          <input
            className='search-contacts'
            type='text'
            placeholder='search'
            value={query}
            onChange={(event) => this.updateQuery(event.target.value)}
          />
        </div>

        {showingContacts.length !== contacts.length && (
          <div className='showing-contacts'>
            <span>Now showing {showingContacts.length} of {contacts.length} total</span>
            <button onClick={this.clearQuery}>Show all</button>
          </div>
        )}

        <ol className='contact-list'>
          {showingContacts.map((contact) => (
            <li key={contact.email} className='contact-list-item'>
            <Avatar>
            <AccountCircle />
            </Avatar>
              <div className='contact-details'>
              <p style={{fontSize: 20}}>{contact.name}</p>
              <p style={{color: grey500}}>{contact.email}</p>
              </div>

              <div className='contact-details'>
              <span style={{color: grey500}}>{moment(contact.lastLogin).format("DD/MM/YYYY")} &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; {moment(contact.lastLogin).fromNow()}</span>
              </div>

              <IconButton aria-label="Actions">
              <MoreVert />
              </IconButton>
            </li>
          ))}
        </ol>
      </div>
    )
  }
}

export default ListContacts
