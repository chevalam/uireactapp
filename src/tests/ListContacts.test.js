import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import ListContacts from '../components/ListContacts';

const contacts = [{}]

configure({ adapter: new Adapter() });

describe('ListContacts', () => {
    it('should render correctly', () => {
        const output = shallow(
            <ListContacts contacts={contacts} />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
