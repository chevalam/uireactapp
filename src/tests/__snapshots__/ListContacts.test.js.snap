// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContacts should render correctly 1`] = `
<div
  className="list-contacts"
>
  <div
    className="list-contacts-top"
  >
    <input
      className="search-contacts"
      onChange={[Function]}
      placeholder="search"
      type="text"
      value=""
    />
  </div>
  <ol
    className="contact-list"
  >
    <li
      className="contact-list-item"
    >
      <Avatar
        size={40}
      >
        <pure(AccountCircle) />
      </Avatar>
      <div
        className="contact-details"
      >
        <p
          style={
            Object {
              "fontSize": 20,
            }
          }
        />
        <p
          style={
            Object {
              "color": "#9e9e9e",
            }
          }
        />
      </div>
      <div
        className="contact-details"
      >
        <span
          style={
            Object {
              "color": "#9e9e9e",
            }
          }
        >
          04/03/2018
                                          
          a few seconds ago
        </span>
      </div>
      <IconButton
        aria-label="Actions"
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        tooltipPosition="bottom-center"
        touch={false}
      >
        <pure(MoreVert) />
      </IconButton>
    </li>
  </ol>
</div>
`;
